src/helpers/clipboard/clipboard_helpers.ts

```diff
@@ -62,22 +65,46 @@ export function getPasteZones<T>(target: Zone[], content: T[][]): Zone[] {
   return target.map((t) => splitZoneForPaste(t, width, height)).flat();
 }
 
-export function parseOSClipboardContent(content: OSClipboardContent): ParsedOSClipboardContent {
-  if (!content[ClipboardMIMEType.Html]) {
+export async function parseOSClipboardContent(
+  env: SpreadsheetChildEnv,
+  content: OSClipboardContent,
+  clipboardId: string
+): Promise<ParsedOSClipboardContent> {
+  let contentClipboardId: string | undefined;
+  let spreadsheetContent: SpreadsheetClipboardData | undefined = undefined;
+  if (content[ClipboardMIMEType.Html]) {
+    const htmlDocument = new DOMParser().parseFromString(
+      content[ClipboardMIMEType.Html],
+      "text/html"
+    );
+    const oSheetClipboardData = htmlDocument
+      .querySelector("div")
+      ?.getAttribute("data-osheet-clipboard");
+    spreadsheetContent = oSheetClipboardData && JSON.parse(oSheetClipboardData);
+    contentClipboardId = spreadsheetContent?.clipboardId;
+  }
+  if (contentClipboardId !== clipboardId) {
+    const clipboardContent: ParsedOSClipboardContent = {
+      text: content[ClipboardMIMEType.PlainText],
+      data: spreadsheetContent,
+    };
+    for (const type of AllowedImageMimeTypes) {
+      if (content[type]) {
+        // TODO: support multiple import
+        try {
+          const imageData = await env.imageProvider?.uploadFile(content[type]!);
```

Comment:
If we paste multiple time the same image we upload it multiple times in the filestroe. Is there any way to fix that ?

Reply:
This is the responsibility of the filestore to check unicity, which is left to the integrator (e.g. the code inside Odoo) to manage duplicates (it could decide to detect already existing file and yield a link against that file). Inside Odoo, we have unicity of the file on disc, just a different attachment link).